asapdiscovery-ml/asapdiscovery/ml/cli.py

```diff
@@ -0,0 +1,467 @@
+)
+@click.option(
+    "--lig", type=bool, help="Include ligand labels as a node attribute information."
+)
+@click.option(
+    "--irreps-edge-attr",
+    type=int,
+    help=(
+        "Which level of spherical harmonics to use for encoding edge attributes "
+        "internally."
+    ),
+)
+@click.option("--num-layers-schnet", type=int, help="Number of network layers.")
+@click.option(
+    "--neighbor-dist",
+    type=float,
+    help="Cutoff distance for including atoms as neighbors.",
+)
+@click.option(
+    "--num-basis",
+    type=int,
+    help="Number of bases on which the edge length are projected.",
+)
+@click.option("--num-radial-layers", type=int, help="Number of radial layers.")
+@click.option(
+    "--num-radial-neurons",
+    type=int,
+    help="Number of neurons in each radial layer.",
+)
+@click.option("--num-neighbors", type=float, help="Typical number of neighbor nodes.")
+@click.option("--num-nodes", type=float, help="Typical number of nodes in a graph.")
+def test(
+    output_dir: Path,
+    model_type: ascfg.ModelType,
+    config_file: Path | None = None,
+    use_wandb: bool = False,
+    sweep: bool = False,
+    wandb_project: str | None = None,
+    wandb_name: str | None = None,
+    extra_config: list[str] | None = None,
+    grouped: bool | None = None,
+    strategy: ascfg.MTENNStrategy | None = None,
+    pred_readout: ascfg.MTENNReadout | None = None,
+    combination: ascfg.MTENNCombination | None = None,
+    comb_readout: ascfg.MTENNReadout | None = None,
+    max_comb_neg: bool | None = None,
+    max_comb_scale: float | None = None,
+    pred_substrate: float | None = None,
+    pred_km: float | None = None,
+    comb_substrate: float | None = None,
+    comb_km: float | None = None,
+    in_feats: int | None = None,
+    num_layers_gat: int | None = None,
+    hidden_feats: str | None = None,
+    num_heads: str | None = None,
+    feat_drops: str | None = None,
+    attn_drops: str | None = None,
+    alphas: str | None = None,
+    residuals: str | None = None,
+    agg_modes: str | None = None,
+    biases: str | None = None,
+    allow_zero_in_degree: bool | None = None,
+    hidden_channels: int | None = None,
+    num_filters: int | None = None,
+    num_interactions: int | None = None,
+    num_gaussians: int | None = None,
+    cutoff: float | None = None,
+    max_num_neighbors: int | None = None,
+    readout: str | None = None,
+    dipole: bool | None = None,
+    mean: float | None = None,
+    std: float | None = None,
+    num_atom_types: int | None = None,
+    irreps_hidden: str | None = None,
+    lig: bool | None = None,
+    irreps_edge_attr: int | None = None,
+    num_layers_schnet: int | None = None,
+    neighbor_dist: float | None = None,
+    num_basis: int | None = None,
+    num_radial_layers: int | None = None,
+    num_radial_neurons: int | None = None,
+    num_neighbors: float | None = None,
+    num_nodes: float | None = None,
+):
```

Comment:
Seeing as there are three semi-orthogonal models each with their own opts, you could do subcommands to separate them, as a whole bunch of commands will be non-compatible and that may not be so obvious

Thinking something like 
```
asap-ml build gat --gat-opt1

...
asap-ml build schnet --schnet-opt1

```

Reply:
is there an easy way to do this in `click`?

Reply:
Yes you can register subcommands, like so 

```python
import click
from gat import gat

@click.group()
def cli():
    pass

cli.add_command(gat)

```

In a different file (or same file) you can have 

```python
import click


@click.group()
def gat():
    pass


@gat.command()
def test():
    click.echo('hello')
```
you can then call it with 

```bash
 scriptname gat test
 >>> hello
```

Reply:
This is pretty non obvious from the `click` docs TBH

Reply:
You have done this super well, well done.